numcodecs/pickles.py

```diff
@@ -48,6 +48,11 @@ def encode(self, buf):
         return pickle.dumps(buf, protocol=self.protocol)
 
     def decode(self, buf, out=None):
+        if PY2:
+            buf = ensure_bytes(buf)
+        else:
+            buf = ensure_contiguous_ndarray(buf)
```

Comment:
Just to provide context from the Python codebase, [Python 2 requires `bytes` here]( https://github.com/python/cpython/blob/d4f9cf5545d6d8844e0726552ef2e366f5cc3abd/Modules/_pickle.c#L1179 ) and [Python 3 uses the (new) buffer protocol to get a C-contiguous array]( https://github.com/python/cpython/blob/d4f9cf5545d6d8844e0726552ef2e366f5cc3abd/Modules/_pickle.c#L1179 ).